### ISSUE TITLE
[pid file mnagement] testHarness suite: random errors due to PID file missing.

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#400]
After re-implementing the test harness engine (PR #394) an issue has been found (described in one of that PR comments):

In some case (randomly) I get the following in some tests (Debian environment):

```
010/194: compound_convop7_xml.test ... (ERROR 7) funcTest.sh/Create and query compound values (EVector XML): (compound_convop7_xml.test): teardown produced output on stderr
cat: /tmp/orion_9999.pid: No such file or directory
make: *** [functional_test] Error 7
```

```
168/194: xmlParse/compound0.test ... (ERROR 7) funcTest.sh/Compound Values for UpdateContextRequest 0: (xmlParse/compound0.test): teardown produced output on stderr
cat: /tmp/orion_9999.pid: No such file or directory
make: *** [functional_test] Error 7
```

Looking at brokerStop() implementation, it is related to the following:

```
kill $(cat $pidFile) 2> /dev/null
```

The situation is weird, as an error in cat means that the broker has already been stopped... If the stopping cause was a crash the test should have probably failed in the output comparation phase, not in the teardown phase.

Any idea of what can be happening in this case?

**Effort: 5 man day**

**Comment by fgalan** [comment]
Changed to nice-to-have (in my opinion, it is not so critical to be "should").

**Comment by fgalan** [comment]
The issue is old (last comment 3 years ago) and the testharness has been heavily rewritten in this period. Thus this issue could be obsolete. We are going to close it. If problems with PID file in testHarness happen again, a new fresh issue will be created.

**Comment by kzangeli** [comment]
Actually, if the broker dies in "an ugly way", the pid file remains in the
file system and a subsequent start of the broker fails.
The solution (only speaking about functests) is to remove the pid file
before starting the broker.
I have this already implemented in the orionld fork
